docs/adr/004_region_as923_implementation.md

```diff
@@ -2,34 +2,119 @@
+
+- End device sends join request using DR2 - DR5 (which always complies with potential dwell time limitations of 400ms)
+- The actual dwell time limitations are communicated by the gateway to the device using a downstream `TxParamSetupReq` MAC Command
+- End device sends `TxParamSetupAns` MAC Command with empty payload as acknowledgement
+- Both end device and gateway adhere to the limitations from there on
+
+With the LoRaWAN specification 1.0.3, the `TxParamSetupReq/Ans` MAC Command exchange [is subject to a bug if the `TxParamSetupAns` is lost][dwell-time-bug], which leaves the end devices unable to receive downlink messages. The bug will only be fixed in LoRaWAN specification version 1.0.4.
+
 This document summarizes decisions taken for the purpose of implementing support for region AS923.
 
 ## In-Scope
 
 - Support for all countries using frequency plan AS923
 - Calculation of RX1 downstream frequencies and data rates
 - Calculation of RX2 receive window
+- Dwell time support
 
 ## Out-of-scope
 
 - MAC commands - support will be added later on as part of [#414](https://github.com/Azure/iotedge-lorawan-starterkit/issues/414)
 - Adaptive Data Rate - support will be added later on as part of [#415](https://github.com/Azure/iotedge-lorawan-starterkit/issues/415)
-- DwellTime - it is not yet clear whether the parameter needs to be set or not, this is tracked in user story [#732](https://github.com/Azure/iotedge-lorawan-starterkit/issues/732)
 - FOpts - it is not yet clear whether this parameter needs to be used, this is tracked for all regions as part of user story [#717](https://github.com/Azure/iotedge-lorawan-starterkit/issues/717)
 
 ## Decision
 
+### Frequency offsets
+
 The `AS923_FREQ_OFFSET` parameter can be calculated based on the channel 0 and channel 1 frequencies in the LoRa Basics Station configuration. The corresponding channels for region AS923 are defined as follows:
 
 `Channel 0 frequency Hz = 923200000 + AS923_FREQ_OFFSET_HZ`
 `Channel 1 frequency Hz = 923400000 + AS923_FREQ_OFFSET_HZ`
 
 Using this formula we will calculate the offset by subtracting 923200000 from the configured channel 0 frequency. We will use the formula for channel 1 frequency to validate the offset value and throw and exception if values are not the same.
 
-In the implementation of region AS923 the frequencies for channel 0 and 1 will be passed to the region-specific constructor where the offset value will be calculated.  
+In the implementation of region AS923 the frequencies for channel 0 and 1 will be passed to the region-specific constructor where the offset value will be calculated.
+
+### Dwell Times
+
+We will support dwell times through the automatic dwell time management process described in the appendix.
```

Comment:
My understanding after yesterday's meeting was that we would implement manual dwell time management?

Reply:
Based on the feedback from @p-schuler we reconsidered, changed the description of automatic dwell time management and propose to use that approach instead.

Reply:
Resolving the comment as a consequence of the approval